data_spec_validator/spec/validators.py

```diff
@@ -449,3 +454,36 @@ def validate(value, extra, data) -> Tuple[bool, Union[Exception, str]]:
         ok = type(value) == str and match_func and match_func(pattern, value)
         info = '' if ok else ValueError(f'{repr(value)} does not match "{error_regex_param}"')
         return ok, info
+
+
+class CondExistValidator(BaseValidator):
+    name = COND_EXIST
+
+    @staticmethod
+    def validate(value, extra, data) -> Tuple[bool, Union[Exception, str]]:
+        allow_unknown = extra.get('ALLOW_UNKNOWN', False)
+        params = extra.get(CondExistValidator.name, {})
+        any_keys = params.get('ANY', [])
+        must_with_keys = params.get('WITH', [])
+        must_without_keys = params.get('WITHOUT', [])
```

Comment:
I don't want to make them constant for now because I'm still trying to give them a better name.

Instead of (`WITH`, `WITHOUT`, `ANY`) ... do you think (`COEXIST`, `EXCLUDE`, `ANY_OF`) is better?

Reply:
ANY is not the scope of COND_EXIST now